lib/mix/tasks/docs.ex

```diff
@@ -141,8 +140,26 @@ defmodule Mix.Tasks.Docs do
     end
   end
 
-  defp normalize_source_beam(options) do
-    Keyword.put_new(options, :source_beam, Mix.Project.compile_path)
+  defp normalize_source_beam(options, config) do
+    compile_path =
+      if Mix.Project.umbrella?(config) do
+        umbrella_compile_paths()
+      else
+        Mix.Project.compile_path
+      end
+
+    Keyword.put_new(options, :source_beam, compile_path)
+  end
+
+  defp umbrella_compile_paths do
```

Comment:
I think this could should be like this:

```
build = Mix.Project.build_path()
for %{app: app} <- Mix.Dep.Umbrella.unloaded do
  Path.join([build, "lib", Atom.to_string(app), "ebin"]
end
```

Also please add a TODO that says: "TODO: Use Mix.Project.apps_path when we require Elixir v1.4+".